libclc/utils/libclc-remangler/LibclcRemangler.cpp

```diff
@@ -501,28 +528,64 @@ class TargetTypeReplacements {
   SmallDenseMap<const char *, const char *> getAliasTypeReplacements() {
     return AliasTypeReplacements;
   }
+
+  SmallDenseMap<const char *, const char *>
+  getRemangledAliasTypeReplacements() {
+    return RemangledAliasTypeReplacements;
+  }
 };
 
-bool createAlias(Module *M, std::string originalMangledName,
-                 const itanium_demangle::Node *functionTree,
-                 TargetTypeReplacements replacements) {
-  Remangler ATR{functionTree, replacements.getAliasTypeReplacements()};
-  std::string RemangledAliasName = ATR.remangle();
+bool createAliasFromMap(
+    Module *M, std::string originalName,
+    const itanium_demangle::Node *functionTree,
+    SmallDenseMap<const char *, const char *> TypeReplacements,
+    bool AliaseeTypeReplacement = false) {
+  Remangler ATR{functionTree, TypeReplacements};
+  std::string RemangledName = ATR.remangle();
 
   if (ATR.hasFailed())
     return false;
 
   // Name has not changed from the original name.
-  if (RemangledAliasName == originalMangledName)
+  if (RemangledName == originalName)
     return true;
 
-  Function *Alias = M->getFunction(RemangledAliasName);
-  if (Alias) {
-    GlobalAlias::create(originalMangledName, Alias);
-  } else if (Verbose) {
-    std::cout << "Could not create alias " << originalMangledName
-              << " : missing " << RemangledAliasName << std::endl;
+  if (AliaseeTypeReplacement) {
+    Function *Aliasee = M->getFunction(RemangledName);
+    if (Aliasee) {
+      GlobalAlias::create(originalName, Aliasee);
+    } else if (Verbose) {
+      std::cout << "Could not create alias " << originalName << " : missing "
+                << RemangledName << std::endl;
+    }
+  } else {
+    Function *Alias = M->getFunction(originalName);
+    if (Alias) {
+      GlobalAlias::create(RemangledName, Alias);
+    } else if (Verbose) {
+      std::cout << "Could not create alias " << RemangledName << " : missing "
+                << originalName << std::endl;
+    }

```

Comment:
Slightly nitpicky, but instead of repeating the code I think it would look cleaner if you make `AliasName` and `AliaseeName` as `StringRef` and set them here, then you only need this code once.

Reply:
That is a good idea, i have updated it, the code looks cleaner.

Reply:
I agree. Thanks!